### PR TITLE
Allow to install the noiseprofile tools in the system

### PR DESCRIPTION
**Pull request by cryptomilk**
cmake -DBUILD_NOISE_TOOLS=ON ..

### CODE REVIEW

File: tools/noise/Makefile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-PROGS = noiseprofile floatdump
```
[the remark this answers is not in the document]
Comment (cryptomilk) [reply tools/noise/Makefile]: If you run the script out of the directory and nothing is installed in the system, dt-gen-noiseprfile will compile the tool automatically, see subr.sh. I did want to break running the code as it was before, it is called backwards compatibility ;-)

[the remark this answers is not in the document]
Comment (darix) [reply tools/noise/Makefile]: Hmm question to houz and hanatos then if they want to keep this backwards compat now that this is properly intergrated.

[the remark this answers is not in the document]
Comment (cryptomilk) [reply tools/noise/Makefile]: Yes, I want backward compatibility. These changes will not be release before Christmas this year and I have to tell people how to do it in my tutorial. I don't want to describe 20 different ways. It should be easy as it was an easier in future.

File: tools/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,5 @@
+project(tools C)
```
Comment (houz) [inline tools/CMakeLists.txt]: Why do you create a new project?

File: tools/noise/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,47 @@
+project(noise-tools C)
```
Comment (houz) [inline tools/noise/CMakeLists.txt]: Why another project?

File: tools/noise/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+find_library(M_LIBRARY m)
```
Comment (houz) [inline tools/noise/CMakeLists.txt]: You can just add "m" to the target_link_libraries call. dt proper does the same, so it should work here, too.

File: tools/noise/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+add_executable(dt-noiseprofile noiseprofile.c)
```
Comment (houz) [inline tools/noise/CMakeLists.txt]: It's "darktable" and not "dt" in executable names.

File: tools/noise/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+target_link_libraries(dt-noiseprofile ${M_LIBRARY})
+
+install(TARGETS
+            dt-noiseprofile
+        DESTINATION
+            ${CMAKE_INSTALL_LIBEXECDIR}/darktable/tools
+        COMPONENT
+            DTApplication)
+
+if (UNIX AND NOT WIN32)
```
Comment (houz) [inline tools/noise/CMakeLists.txt]: I guess we can move the whole content of the file into this check. Why would we install a helper tool when the main program is missing?
That being said, `UNIX AND NOT WIN32` sounds wrong.

File: tools/noise/CMakeLists.txt [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    install(PROGRAMS
+                dt-gen-noiseprofile
```
Comment (houz) [inline tools/noise/CMakeLists.txt]: `s/dt/darktable/`

File: tools/noise/CMakeLists.txt [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+            else ()
+                execute_process(COMMAND \"${CMAKE_COMMAND}\" -E create_symlink
+                                ${_filepath}
+                                \$ENV{DESTDIR}/${_sympath})
+            endif ()
+        ")
+        install(CODE "message(\"-- Creating symlink: ${_sympath} -> ${_filepath}\")")
+    endmacro(install_symlink)
+
+    install_symlink("${CMAKE_INSTALL_FULL_LIBEXECDIR}/darktable/tools/dt-gen-noiseprofile" "${CMAKE_INSTALL_FULL_BINDIR}/dt-gen-noiseprofile")
```
Comment (houz) [inline tools/noise/CMakeLists.txt]: What is that good for? To have the main script in people's $PATH?

File: tools/noise/Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (houz) [reply tools/noise/Makefile]: Drop the Makefile if you add a CMakeLists.txt that does the same thing.

File: tools/noise/dt-gen-noiseprofile [hunk 9]
```diff
@@ -2,8 +2,21 @@
...
+	script=$(readlink ${script})
+	count=$(expr ${count} + 1)
+	if [ ${count} -gt 100 ]
+	then
+		echo "Too many symbolic links"
+		exit 1
+	fi
+done
+scriptdir=$(dirname ${script})
+scriptname=$(basename ${script})
```
Comment (houz) [inline tools/noise/dt-gen-noiseprofile]: Just don't link the script to /usr/bin/ and you don't need this.

File: tools/noise/dt-gen-noiseprofile [hunk 10]
```diff
@@ -61,6 +74,11 @@ if [ "$iso_settings" ]; then
 	iso_settings=$(sort_iso_list $iso_settings)
 fi
 
+noiseprofile="$(which dt-noiseprofile || true)"
```
Comment (houz) [inline tools/noise/dt-gen-noiseprofile]: That tool will never be in PATH.

File: tools/noise/dt-gen-noiseprofile [hunk 11]
```diff
@@ -98,6 +119,11 @@ if [ "$missing_tool" = "1" ]; then
 	exit 1
 fi
 
+# Check if we have compiled dt-noiseprofile
```
Comment (houz) [inline tools/noise/dt-gen-noiseprofile]: Why?

File: tools/CMakeLists.txt (hunk 2 above)
Comment (cryptomilk) [reply tools/CMakeLists.txt]: Normally I always do that. project() gives you some nice variables <project>_SOURCE_DIR and <project>_BINARY_DIR which are quite handy if you have headers in that directory and you need to include them somewhere else. However we don't use it so I can remove it.

File: tools/noise/CMakeLists.txt (hunk 3 above)
Comment (cryptomilk) [reply tools/noise/CMakeLists.txt]: Normally I always do that. project() gives you some nice variables <project>_SOURCE_DIR and <project>_BINARY_DIR which are quite handy if you have headers in that directory and you need to include them somewhere else. However we don't use it so I can remove it.

File: tools/noise/CMakeLists.txt (hunk 6 above)
Comment (cryptomilk) [reply tools/noise/CMakeLists.txt]: Nope, MinGW is UNIX and WIN32 ;-)

File: tools/noise/CMakeLists.txt (hunk 8 above)
Comment (cryptomilk) [reply tools/noise/CMakeLists.txt]: I can remove that if you don't want that.

File: tools/noise/dt-gen-noiseprofile (hunk 9 above)
Comment (cryptomilk) [reply tools/noise/dt-gen-noiseprofile]: Ok, will remove it.

File: tools/noise/dt-gen-noiseprofile (hunk 11 above)
Comment (cryptomilk) [reply tools/noise/dt-gen-noiseprofile]: I wanted to keep backwards compatibility, but then I will remove it.

**Comment by cryptomilk** [comment]
Updated ...

**Pull request merged**